Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.mm

```diff
@@ -277,6 +277,45 @@
     return CFDataGetBytePtr(m_data.get()) + sourcePosition;
 }
 
+bool PDFPluginBase::getByteRanges(CFMutableArrayRef dataBuffersArray, const CFRange* ranges, size_t count) const
```

Comment:
Is this ever called with a single range that covers all of `m_data`? If so, should we optimize that case?